integration/tests/conftest.py

```diff
@@ -131,51 +150,59 @@ def bank_account(pytestconfig: Config) -> typing.Optional[Keypair]:
 
 
 @pytest.fixture(scope="session")
-def eth_bank_account(pytestconfig: Config, web3_client) -> typing.Optional[Keypair]:
+def eth_bank_account(pytestconfig: Config, web3_client_session) -> tp.Optional[Keypair]:
     account = None
     if pytestconfig.environment.eth_bank_account != "":
-        account = web3_client.eth.account.from_key(pytestconfig.environment.eth_bank_account)
+        account = web3_client_session.eth.account.from_key(pytestconfig.environment.eth_bank_account)
     yield account
 
 
 @pytest.fixture(scope="session")
-def solana_account(bank_account, pytestconfig: Config, sol_client):
+def solana_account(bank_account, pytestconfig: Config, sol_client_session):
     account = Keypair.generate()
     if pytestconfig.environment.use_bank:
-        sol_client.send_sol(bank_account, account.public_key, int(0.5 * LAMPORT_PER_SOL))
+        sol_client_session.send_sol(bank_account, account.public_key, int(0.5 * LAMPORT_PER_SOL))
     else:
-        sol_client.request_airdrop(account.public_key, 1 * LAMPORT_PER_SOL)
+        sol_client_session.request_airdrop(account.public_key, 1 * LAMPORT_PER_SOL)
     yield account
     if pytestconfig.environment.use_bank:
-        balance = sol_client.get_balance(account.public_key, commitment=commitment.Confirmed).value
+        balance = sol_client_session.get_balance(account.public_key, commitment=commitment.Confirmed).value
         try:
-            sol_client.send_sol(account, bank_account.public_key, balance - 5000)
+            sol_client_session.send_sol(account, bank_account.public_key, balance - 5000)
         except:
             pass
 
 
+@pytest.fixture(scope="class")
+def accounts(request, accounts_session):
+    if inspect.isclass(request.cls):
+        request.cls.accounts = accounts_session
+    yield accounts_session
+    accounts_session._accounts = []
```

Comment:
What do you need  to make it empty for?

Reply:
To handle situation from base classes, when all accounts was class based. It will changes in https://neonlabs.atlassian.net/browse/NDEV-2493